alf/pretrained_models/pretrained_model.py

```diff
@@ -0,0 +1,96 @@
+# Copyright (c) 2023 Horizon Robotics and Hobot Contributors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+import torch.nn as nn
+
+from typing import Callable
+
+from .model_adapters import LoRA
+
+
+class PretraindModel(nn.Module):
+    """A wrapper class for managing pretrained models.
+
+    A pretrained model is generally large and its weights will always be frozen.
+    For finetuning, we can add small adapters to some of its layers and only the
+    adapter weights will be trained on downstream tasks. See
+    `<https://docs.adapterhub.ml/methods.html>`_.
+    """
+
+    def __init__(self, model, name: str = 'PretrainedModel'):
+        super().__init__()
+        # Freeze all parameters
+        for para in model.parameters():
+            para.requires_grad = False
+        self._model = model
+        self._adapters = nn.ModuleList()
+        self._name = name
+
+    def add_adapter(self, adapter_cls: Callable = LoRA):
```

Comment:
> I am a little worried about the side effect introduced by having `add_adapter`. With this, the user can call `add_adapter` flexibly and maybe even after a checkpoint is saved. This also requires the consumer of the lora weights have to call `add_adapter`s exactly the same as when the the lora weights are trained.
> 
> An alternative would be to pass the adapter constructors as arguments to the constructor. In this way, it is easily configurable with `alf.config`, `partial` or subclass, so that as long as the consumer and producer get the same configuration, the checkpoints are sure to be compatible.

Moved adapter cls list to the constructor of `PretrainedModel`